bin/test/main.ml

```diff
@@ -250,13 +250,10 @@ let update_file_or_block ?root ppf md_file ml_file block direction =
   in
   let direction =
     match direction with
-    | `To_md -> `To_md
-    | `To_ml -> `To_ml
-    | `Infer_timestamp ->
-       let md_file_mtime = (Unix.stat md_file).st_mtime in
-       let ml_file_mtime = (Unix.stat ml_file).st_mtime in
-       if ml_file_mtime < md_file_mtime then `To_ml
-       else `To_md
+    | Some `To_md -> `To_md
+    | Some `To_ml -> `To_ml
+    | None ->
+        Fmt.failwith "--direction must be set to update this file."
```

Comment:
I like that it makes `--direction` optional!

Do you think we could include the location of the block in the markdown file that requires the option to be set? The filename at least would be great. I also think the error message could be slightly more explicit, what would you think about somethng like: "This block must be synchronised with <ml_file>, you must set a promotion direction for <md_file>. See --direction documentation." ?

Reply:
Improving errors need a bit of work, I'll submit an other PR for easier reviews. For the tests too.
The message can be improved.

Reply:
Ok, let's leave the location out for now. Could you, as part of this PR, improve the error message a little bit and still add a test just to make sure the error is properly triggered and ensure we don't inadvertently break it in the future?

I believe this can be done quite easily as both file names are available in the scope at this point.

Reply:
I changed the message.
Location/testing in https://github.com/realworldocaml/mdx/pull/172. Not done in this PR because the output would not looks good (uncaught exception) and should be discussed separately.

Reply:
I'm not sure I understand how that prevents you from adding a test to that PR, especially since the first commit in #172 is pretty much what I'd expect from this PR.

Reply:
Currently the output is an uncaught exception message from OCaml's runtime, which is not nice. But more importantly, adding this test requires to add a mechanism to run it that may require discussion.
Would you prefer if I reorder the PRs ? #172 first and this one rebased on it ?